util/asm_dev/armv8/p256/src/main.c

```diff
@@ -0,0 +1,212 @@
+     * Test select_w5
+     */
+    passed = 1;
+    for(j = 1; j <= TABLE_LONG_SIZE; j++)
+    {
+      table[j-1] = (uint64_t)j << 2;
+    }
+
+    for(index = 1; index <= NUM_ENTRIES; index ++)
+    {
+      select_w5(val, table, index);
+
+      // Check the correct entry was selected
+      for(j = 0; j < ENTRY_LONG_SIZE; j++)
+      {
+        // Calculating the index to the 64-bit part of the table entry
+        idx_64 = ((index-1) * ENTRY_LONG_SIZE) + (uint32_t) j;
+        if(val[j] != table[idx_64])
+        {
+          printf("Error: difference at index: %d, j: %d, idx_64: %d, val[j]:%lX, table[idx_64]: %lX\n",
+                 index, j, idx_64, val[j], table[idx_64]);
+          passed = 0;
+        }
+      }
+    }
+    if (1 == passed)
+    {
+        printf("select_w5 Test passed\n");
+    }
+
+    /*
+     * Test select_w7
+     */
+    passed = 1;
+    for(j = 1; j <= TABLE_LONG_SIZE_W7; j++)
+    {
+      table_w7[j-1] = (uint64_t)j << 2;
+    }
+
+    for(index = 1; index <= NUM_ENTRIES_W7; index ++)
+    {
+      select_w7(val_w7, table_w7, index);
+
+      // Check the correct entry was selected
+      for(j = 0; j < ENTRY_LONG_SIZE_W7; j++)
+      {
+        // Calculating the index to the 64-bit part of the table entry
+        idx_64 = ((index-1) * ENTRY_LONG_SIZE_W7) + (uint32_t) j;
+        if(val_w7[j] != table_w7[idx_64])
+        {
+          printf("Error (w7): difference at index: %d, j: %d, idx_64: %d, val[j]:%lX, table[idx_64]: %lX\n",
+                 index, j, idx_64, val_w7[j], table_w7[idx_64]);
+          passed = 0;
+        }
+      }
+    }
+    if (1 == passed)
+    {
+        printf("select_w7 Test passed\n");
+    }
+
+#endif
+
+#ifdef BEEU
+    /*
+     * Test beeu
+     */
+
+    printf("a = 2\n");
+    a[0] = 2;
+    a[1] = a[2] = a[3] = 0;
+
+    beeu_res = beeu_mod_inverse_vartime(out, a, order_words);
+    beeu_print(beeu_res, out_exp_2, out);
+
+    printf("a = 1\n");
+    a[0] = 1;
+    out_exp[0] = 1;
+    out_exp[1] = out_exp[2] = out_exp[3] = 0;
+
+    beeu_res = beeu_mod_inverse_vartime(out, a, order_words);
+    beeu_print(beeu_res, out_exp, out);
+
+    for (k = 1; k < K_MAX; k++)
+    {
+        a[0] = k;
+        if (k >= (K_MAX >>1))
+        {
+            a[1] = k << 8;
+            a[2] = k << 32;
+            a[3] = k << 48;
+        } else {
+            a[1] = a[2] = a[3] = 0;
+        }
+        // out = a^{-1} mod n
+        beeu_res = beeu_mod_inverse_vartime(out, a, order_words);
+        if (beeu_res == 1)
+        {
+            // a_out = out^{-1} mod n
+            beeu_res = beeu_mod_inverse_vartime(a_out, out, order_words);
+            if (beeu_res != 1    ||
+                a_out[0] != a[0] ||
+                a_out[1] != a[1] ||
+                a_out[2] != a[2] ||
+                a_out[3] != a[3]
+                )
+            {
+                printf("beeu Test FAILED at k = %ld; a_out != a\n", k);
+            }
+        }
+    }
+    if (k == K_MAX)
```

Comment:
Won't this conditional always be true? The for-loop runs until `k = K_MAX`.

Reply:
That's right. I added a missing `break` statement in the loop above. I had removed it while debugging.